### PR TITLE
Add daml clean command

### LINKED ISSUE #1223: `damlc clean` command to clean up package database and dist folder.
For #644 I was thinking of implementing a `daml clean` command in daml-helper but this really belongs in `damlc`, which generates all the artifacts. Then once it's available it can be exposed in the assistant through the usual means (changing `releases/sdk-config.yaml`).

This command should remove any build artifacts -- `.package-db` and `dist/`.

It should fail if run outside a project. It should succeed if there are no build artifacts present (like `rm -f`).

cc @robin-da

### PR DESCRIPTION
**Pull request by associahedron**
Adds a `daml clean` to delete build artifacts, and an integration test to ensure it doesn't drift out of sync with `daml build`. Fixes #1223

### CODE REVIEW

File: daml-assistant/integration-tests/src/Main.hs [hunk 1]
```diff
@@ -213,6 +215,36 @@ quickstartTests quickstartDir mvnDir = testGroup "quickstart" $
     where
         mvnRepoFlag = "-Dmaven.repo.local=" <> mvnDir
 
+-- | Ensure that daml clean removes precisely the files created by daml build.
+cleanTests :: FilePath -> TestTree
```
Comment (cocreature) [inline daml-assistant/integration-tests/src/Main.hs]: Nice, clever way to test this!

File: daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs [hunk 2]
```diff
@@ -404,8 +412,22 @@ createProjectPackageDb lfVersion fps = do
...
+
+-- | Remove any build artifacts if they exist.
+execClean :: IO ()
+execClean = do
+    withProjectRoot $ \_relativize -> do
+        isProject <- doesFileExist projectConfigName
+        if isProject then do
+            removePathForcibly projectPackageDatabase
+            removePathForcibly ".interfaces"
+            removePathForcibly "dist"
```
Comment (neil-da) [inline daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs]: Would prefer if we put everything in one dir...

Comment (neil-da) [reply daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs]: Even better if the dir was `.daml`.

Comment (associahedron) [reply daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs]: Me too. I think that can be done in a separate PR though.

**Pull request merged**